src/Analyzer/ReferenceTrimmerAnalyzer.cs

```diff
@@ -90,7 +91,18 @@ private static void DumpUsedReferences(CompilationAnalysisContext context)
             }
         }
 
-        WriteUsedReferencesToFile(usedReferences, declaredReferencesPath);
+        HashSet<string> unusedReferences = new(StringComparer.OrdinalIgnoreCase);
+        foreach (MetadataReference metadataReference in compilation.References)
+        {
+            if (metadataReference.Display != null)
+            {
+                unusedReferences.Add(metadataReference.Display);
+            }
+        }
+
+        unusedReferences.ExceptWith(usedReferences);
+
+        DumpReferencesInfo(usedReferences, unusedReferences, declaredReferencesPath);
```

Comment:
This should be being an env var to avoid the perf hit the vast majority of the time.

Reply:
There's another way to pass MSBuild properties to an analyzer's global options through `CompilerVisibleProperty`